### PR TITLE
Use/party

### PR DESCRIPTION
**Pull request by bazimov**

### CODE REVIEW

File: src/lavatory/utils/artifactory.py [hunk 1]
```diff
@@ -1,100 +1,92 @@
...
+from ..credentials import load_credentials
 
-    def delete(self, path):
-        LOG.info("DELETE", path)
-        return
+LOG = logging.getLogger(__name__)
 
-        """
-        response = requests.delete(baseurl + "/" + path)
+CREDENTIALS = load_credentials()
```
Comment (ndcampbell) [inline src/lavatory/utils/artifactory.py]: I think this should happen inside the class when it is initialized.

File: src/lavatory/utils/artifactory.py [hunk 2]
```diff
@@ -1,100 +1,92 @@
...
 
-        if response.ok:
-            return response.json()
-        else:
-            response.raise_for_status()
-            #raise Exception("HTTP DELETE {}: Server code {}".format(endpoint, response.status_code))
-        """
+PARTY_CONFIG = {
```
Comment (ndcampbell) [inline src/lavatory/utils/artifactory.py]: Same with this, I think this should happen inside the class when it is initialized.

File: src/lavatory/utils/artifactory.py [hunk 3]
```diff
@@ -1,100 +1,92 @@
...
+    'artifactory_url': CREDENTIALS['artifactory_url'],
+    'username': CREDENTIALS['artifactory_username'],
+    'password': base64.encodebytes(bytes(CREDENTIALS['artifactory_password'], 'utf-8')),
+}
 
 
 class Artifactory(object):
 
-    def __init__(self, baseurl, user, passwd):
```
Comment (ndcampbell) [inline src/lavatory/utils/artifactory.py]: We are still going to want an `init`, but it should just take in `repo_name` and setup credentials. Then we dont have to pass repo_name to the other functions

File: src/lavatory/utils/artifactory.py (hunk 1 above)
Comment (bazimov) [reply src/lavatory/utils/artifactory.py]: Moved inside the __init__.

File: src/lavatory/utils/artifactory.py (hunk 2 above)
Comment (bazimov) [reply src/lavatory/utils/artifactory.py]: Moved inside the __init__.

File: src/lavatory/utils/artifactory.py (hunk 3 above)
Comment (bazimov) [reply src/lavatory/utils/artifactory.py]: added repo_name with default None.

File: src/lavatory/utils/artifactory.py [hunk 4]
```diff
@@ -1,100 +1,90 @@
...
-            if repo["repoKey"] == "TOTAL" :
-                continue
-                
-            if not reponame or reponame == repo["repoKey"]:
-                repos[repo["repoKey"]] = repo
-    
-        return repos
-    
-    def purge(self, repo, dryrun, artifacts):
+    def purge(self, repo, dry_run, artifacts):
```
Comment (ndcampbell) [inline src/lavatory/utils/artifactory.py]: repo can be removed from this function and use`self.repo_name`

File: src/lavatory/utils/artifactory.py [hunk 5]
```diff
@@ -1,100 +1,90 @@
...
-                    self.http.delete(artifact)
+                    self.artifactory.request(artifact, method='delete')
                     purged += 1
                 except Exception as e:
                     LOG.error(str(e))
 
         return purged
 
-    def filter(self, repo, terms=[], depth=3, sort=None, offset=None, limit=None):
+    def filter(self, repo, terms=None, depth=3, sort=None, offset=None, limit=None):
```
Comment (ndcampbell) [inline src/lavatory/utils/artifactory.py]: repo can be removed from this function and use`self.repo_name`

File: src/lavatory/utils/artifactory.py [hunk 6]
```diff
@@ -112,51 +102,61 @@ def filter(self, repo, terms=[], depth=3, sort=None, offset=None, limit=None):
...
-    
-        response = self.http.post("/search/aql", aql)
+
+        response = self.artifactory.find_by_aql(criteria=aql)
 
         return response["results"]
-    
-    def retain(self, repo, specproject, depth, terms=None, count=None, weeks=None):
+
+    def retain(self, repo, spec_project, depth, terms=None, count=None, weeks=None):
```
Comment (ndcampbell) [inline src/lavatory/utils/artifactory.py]: repo can be removed from this function and use`self.repo_name`

Comment (bazimov) [reply src/lavatory/utils/artifactory.py]: changed to repo_name from init.

File: src/lavatory/utils/artifactory.py (hunk 5 above)
Comment (bazimov) [reply src/lavatory/utils/artifactory.py]: changed to repo_name from init.

File: src/lavatory/utils/artifactory.py (hunk 4 above)
Comment (bazimov) [reply src/lavatory/utils/artifactory.py]: changed to repo_name from init.

File: src/lavatory/utils/artifactory.py [hunk 7]
```diff
@@ -112,51 +103,63 @@ def filter(self, repo, terms=[], depth=3, sort=None, offset=None, limit=None):
...
-        if weeks:
-            now = datetime.datetime.now()
-            before = now - datetime.timedelta(weeks=weeks)
-            created = before.strftime("%Y-%m-%dT%H:%M:%SZ")
-
         purgable = []
 
-        for project in self.filter(repo, depth=depth):
-            if specproject and specproject!=project["name"]:
+        for project in self.filter(self.repo_name, depth=depth):
```
Comment (ndcampbell) [inline src/lavatory/utils/artifactory.py]: Dont need to pass in `self.repo_name` since it is already part of the class

Comment (bazimov) [reply src/lavatory/utils/artifactory.py]: ah missed that one. removed.

**Comment by ndcampbell** [comment]
@bazimov last thing to point out, `lavatory purge` isnt working right now with the default policy since things were changed. I think that should be updated in this PR too.

My opinion is that at https://github.com/gogoair/lavatory/blob/master/src/lavatory/commands/purge.py#L28 you create another artifactory object with `repo_name=repo`, and then you pass that to the `policy.purgelist(artifactory)`

The policy then would look like:

```def purgelist(artifactory):
    return artifactory.retain(None, depth=2, terms=[
            { "$or": [
                ]}
        ])
```

That `None` is still the project setting, which I think should be abstracted out later

**Comment by bazimov** [comment]
@ndcampbell did you mean like last commit? I still cannot make that find_by_aql work. Looking into it.

**Comment by ndcampbell** [comment]
@bazimov yup! Looks good. I'll poke around at the find_by_aql issue too.

**Comment by ndcampbell** [comment]
@bazimov figured it out. Couple of issues. That `find_by_aql` expects a dict, we are sending a string. We need to just remove the `json.dumps` from `find_expr` and then set `find_by_aql(criteria=find_expr)`

you can delete the `sort,` `offset`, and `limit` parts too, those are all in the party function: https://github.com/tedops/party/blob/master/party/aql.py

**Comment by ndcampbell** [comment]
@bazimov this looks good to me, are you planning on adding anything else or is this still WIP?

**Pull request merged**